peer.go

```diff
@@ -76,9 +76,20 @@ func newPeerList(root *RootPeerList) *PeerList {
 	}
 }
 
-// SetStrategy sets customized peer selection stratedgy.
+// SetStrategy sets customized peer selection strategy.
 func (l *PeerList) SetStrategy(sc ScoreCalculator) {
+	l.Lock()
+	defer l.Unlock()
```

Comment:
hmm, it's [there](https://github.com/uber/tchannel-go/pull/583/commits/aa93e59c2ced3b4ed79de3e5e0274e194891be84#diff-32e1ab53c69bf3272bd9e4b51b9bb105R83)